### ISSUE TITLE
insightface model is required for FaceID models

### ISSUE THREAD

**Issue by YoucanBaby** [cubiq/ComfyUI_IPAdapter_plus#362]
Dear developer,

When I run **ipadapter-faceid-pluv2_sd15**, I get the error: 

```
Error occurred when executing IPAdapterAdvanced:

insightface model is required for FaceID models

File "/root/data/xyf/personalization/ComfyUI/execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "/root/data/xyf/personalization/ComfyUI/execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "/root/data/xyf/personalization/ComfyUI/execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "/root/data/xyf/personalization/ComfyUI/custom_nodes/ComfyUI_IPAdapter_plus/IPAdapterPlus.py", line 580, in apply_ipadapter
return (ipadapter_execute(model.clone(), ipadapter_model, clip_vision, **ipa_args), )
File "/root/data/xyf/personalization/ComfyUI/custom_nodes/ComfyUI_IPAdapter_plus/IPAdapterPlus.py", line 179, in ipadapter_execute
raise Exception("insightface model is required for FaceID models")
```

This is [my workflow](https://github.com/cubiq/ComfyUI_IPAdapter_plus/files/14801843/workflowIpAdapterFixed.json) :

![image](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/43912540/9a8ddbba-2ade-4ac4-93ad-9aa4aa89cea5)

Can you help me?

Best wishes!

**Comment by LukeG89** [comment]
You are using `IPAdapter Advanced` instead of `IPAdapter FaceID`

**Comment by YoucanBaby** [comment]
> You are using `IPAdapter Advanced` instead of `IPAdapter FaceID`
> 
> Also, you don't need to use any other loaders when using the Unified one

Thank you for your reply.

This workflow is a little more complicated. 
Specifically, it use the portrait as the ID feature and the image in the upper right corner as the style feature. 
Therefore, I use two IPAdapters.

**Comment by LukeG89** [comment]
I meant that FaceID only works with IPAdapeter FaceID and not with other IPAdapters. In order to make it work you have to replace that node

**Comment by YoucanBaby** [comment]
> I meant that FaceID only works with IPAdapeter FaceID and not with other IPAdapters. In order to make it work you have to replace that node

I'm currently using a non-FaceID.safetensor for IPAdapter Advanced.
Also, other loaders in Unified have been removed.
The entire process now looks like this:

![image](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/43912540/5f6db557-f44b-41fa-a0d1-bad9167ab7bc)

Now a new problem has arisen, I now get "Segmentation fault (core dumped)" on linux.

Please help me more, I'm a newbie in comfyUI. 

Thank you very much.

**Comment by LukeG89** [comment]
When using normal IPAdapters you have to prepare the image for Clip Vision. Clip Vision wants 224x224 square images, so you can both use `Prep Image for Clip Vision` or resize and rescale the image. I think you get this error because of that. If you want to keep the rectangular image, try using `IPAdaper Tiled`, this one accepts non-squared images. If you use this node, the final image has to be in a similar aspect ratio of the reference.

So, try for example this setup:
![faceid+tiled](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/161304036/1a0dfd74-bcbd-407d-96a3-b1bab49832ea)

**Comment by YoucanBaby** [comment]
> 使用普通 IPAdapter 时，您必须为 Clip Vision 准备图像。 Clip Vision 需要 224x224 方形图像，因此您可以使用图像`Prep Image for Clip Vision`或调整图像大小和比例。我认为您因此而收到此错误。 如果你想保留矩形图像，请尝试使用`IPAdapter Tiled`，这个接受非方形图像。如果使用此节点，最终图像必须具有与参考相似的长宽比。
> 
> 因此，请尝试以下设置： ![Faceid+平铺](https://private-user-images.githubusercontent.com/161304036/318030143-1a0dfd74-bcbd-407d-96a3-b1bab49832ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5MzQ3NDIsIm5iZiI6MTcxMTkzNDQ0MiwicGF0aCI6Ii8xNjEzMDQwMzYvMzE4MDMwMTQzLTFhMGRmZDc0LWJjYmQtNDA3ZC05NmEzLWIxYmFiNDk4MzJlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwMTIwNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTcyN2NiYzZhMTE5MTY0OGY3NzlhZDBiZTYxZmJjOWJhOThmNTdkMDAwMmQ5NDViM2NiN2NlNDQzYzMxNGU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TJq_SmHIm6jRlf2EL-68_3IV7iGDp6qb_x8_2MBtjKk)



> 使用普通 IPAdapter 时，您必须为 Clip Vision 准备图像。 Clip Vision 需要 224x224 方形图像，因此您可以使用图像`Prep Image for Clip Vision`或调整图像大小和比例。我认为您因此而收到此错误。 如果你想保留矩形图像，请尝试使用`IPAdapter Tiled`，这个接受非方形图像。如果使用此节点，最终图像必须具有与参考相似的长宽比。
> 
> 因此，请尝试以下设置： ![Faceid+平铺](https://private-user-images.githubusercontent.com/161304036/318030143-1a0dfd74-bcbd-407d-96a3-b1bab49832ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5MzQ3NDIsIm5iZiI6MTcxMTkzNDQ0MiwicGF0aCI6Ii8xNjEzMDQwMzYvMzE4MDMwMTQzLTFhMGRmZDc0LWJjYmQtNDA3ZC05NmEzLWIxYmFiNDk4MzJlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwMTIwNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTcyN2NiYzZhMTE5MTY0OGY3NzlhZDBiZTYxZmJjOWJhOThmNTdkMDAwMmQ5NDViM2NiN2NlNDQzYzMxNGU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TJq_SmHIm6jRlf2EL-68_3IV7iGDp6qb_x8_2MBtjKk)

Thanks a lot for your patient reply.
I fix this error by install `torch==2.0.0`. I was using `torch==2.2.0` before.
The overall pipeline is no problem.

**Comment by YoucanBaby** [comment]
> When using normal IPAdapters you have to prepare the image for Clip Vision. Clip Vision wants 224x224 square images, so you can both use `Prep Image for Clip Vision` or resize and rescale the image. I think you get this error because of that. If you want to keep the rectangular image, try using `IPAdapter Tiled`, this one accepts non-squared images. If you use this node, the final image has to be in a similar aspect ratio of the reference.
> 
> So, try for example this setup: ![faceid+tiled](https://private-user-images.githubusercontent.com/161304036/318030143-1a0dfd74-bcbd-407d-96a3-b1bab49832ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5MzgwOTQsIm5iZiI6MTcxMTkzNzc5NCwicGF0aCI6Ii8xNjEzMDQwMzYvMzE4MDMwMTQzLTFhMGRmZDc0LWJjYmQtNDA3ZC05NmEzLWIxYmFiNDk4MzJlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwMjE2MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZWYyODkyNDc3NjZiZDhmNDMzMzcwNjJhYmMwOGMwYzJiZmJkNDVhYTQ4MGIxMTMwNjliODg2MTA4OWM3Y2IxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ftLtoixyAYldeXNgSeXe31PwIk1bNrP9cMGr1DhqYTw)


Thanks a lot for your patient reply.
I fix this error by install `torch==2.0.0`. I was using `torch==2.2.0` before.
The overall pipeline is no problem.

**Comment by GenshiStudio** [comment]
![Image](https://github.com/user-attachments/assets/74f881c1-5136-46b5-968e-0a5724911ea7)

I have the same problem
